examples/water/main.rs

```diff
@@ -0,0 +1,797 @@
+                cull_mode: wgpu::CullMode::None,
+                depth_bias: 0,
+                depth_bias_slope_scale: 0.0,
+                depth_bias_clamp: 0.0,
+            }),
+            // What kind of data are we passing in?
+            primitive_topology: wgpu::PrimitiveTopology::TriangleList,
+            // Describes how the colour will be interpolated
+            // and assigned to the output attachment.
+            color_states: &[wgpu::ColorStateDescriptor {
+                format: sc_desc.format,
+                color_blend: wgpu::BlendDescriptor {
+                    src_factor: wgpu::BlendFactor::SrcAlpha,
+                    dst_factor: wgpu::BlendFactor::OneMinusSrcAlpha,
+                    operation: wgpu::BlendOperation::Add,
+                },
+                alpha_blend: wgpu::BlendDescriptor {
+                    src_factor: wgpu::BlendFactor::One,
+                    dst_factor: wgpu::BlendFactor::One,
+                    operation: wgpu::BlendOperation::Max,
+                },
+                write_mask: wgpu::ColorWrite::ALL,
+            }],
+            // Describes how us writing to the depth/stencil buffer
+            // will work. Since this is water, we need to read from the
+            // depth buffer both as a texture in the shader, and as an
+            // input attachment to do depth-testing. We don't write, so
+            // depth_write_enabled is set to false. This is called
+            // RODS or read-only depth stencil.
+            depth_stencil_state: Some(wgpu::DepthStencilStateDescriptor {
+                // We don't use stencil.
+                format: wgpu::TextureFormat::Depth32Float,
+                depth_write_enabled: false,
+                depth_compare: wgpu::CompareFunction::Less,
+                stencil_front: wgpu::StencilStateFaceDescriptor::IGNORE,
+                stencil_back: wgpu::StencilStateFaceDescriptor::IGNORE,
+                stencil_read_mask: 0,
+                stencil_write_mask: 0,
+            }),
+            // Layout of our vertices. This should match the structs
+            // which are uploaded to the GPU. This should also be
+            // ensured by tagging on either a `#[repr(C)]` onto a
+            // struct, or a `#[repr(transparent)]` if it only contains
+            // one item, which is itself `repr(C)`.
+            vertex_state: wgpu::VertexStateDescriptor {
+                // We don't actually use indices, since it's unnecessary
+                // because we duplicate all the data anyway. This is
+                // necessary to achieve the low-poly effect.
+                index_format: wgpu::IndexFormat::Uint16,
+                vertex_buffers: &[
+                    wgpu::VertexBufferDescriptor {
+                        stride: water_vertex_size as wgpu::BufferAddress,
+                        step_mode: wgpu::InputStepMode::Vertex,
+                        attributes: &[
```

Comment:
did you consider using the `vertex_attr_array` macro like some of the other examples?